reco_utils/dataset/spark_splitters.py

```diff
@@ -3,8 +3,18 @@
 
 import numpy as np
 
-from pyspark.sql import Window
-from pyspark.sql.functions import col, row_number, broadcast, rand, collect_list, size
+try:
+    from pyspark.sql import Window
+    from pyspark.sql.functions import (
+        col,
+        row_number,
+        broadcast,
+        rand,
+        collect_list,
+        size,
+    )
+except ImportError:
+    pass  # skip this import if we are in pure python environment
```

Comment:
when is this imported in a non-spark environment? i would think it would be better to raise the error at that time?

Reply:
if we don't add this clause, we can't run the cpu or gpu tests, because it will raise an error when trying to import spark